include/range/v3/view/any_view.hpp

```diff
@@ -410,18 +493,22 @@ namespace ranges
                 {
                     return detail::make_unique<any_view_impl>(range_box_t::get());
                 }
+                std::size_t size() const // override-ish
+                {
+                    return static_cast<std::size_t>(ranges::size(range_box_t::get()));
+                }
             };
         }
         /// \endcond
 
         /// \brief A type-erased view
         /// \ingroup group-views
-        template<typename Ref, category Cat = category::input>
+        template<typename Ref, category Cat = category::input, typename enable = void>
```

Comment:
Can you rewrite this so that the SFINAE parameter is not part of the public type?

Reply:
Not, trivially, no. The obvious solution would be to rename the class, move it to detail and define an alias template. This works, but then we loose the type deduction guide in C++17.
Instead of an alias I can inherit, but inheriting the templetized constructors doesn't work well, forcing me to move logic (i.e. full constructor definitions)  into the new class, duplicating code and further increasing the level of non-trivial inheritance and template specialisation.
But maybe I am overlooking something?